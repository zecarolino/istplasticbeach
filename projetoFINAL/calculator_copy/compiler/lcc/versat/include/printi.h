#ifndef _PRINTI_H_
#define _PRINTI_H_

#include "mod.h"
#include "div.h"
#include "putchar.h"

void printi(int n, int b) {
	int a, c;
	if (a = n / b) printi(a, b);
	c = n % b;
	if (c <= 9) putchar(c + '0');
	else putchar(c + 'A' -10);
}

#endif /* _PRINTI_H_ */
