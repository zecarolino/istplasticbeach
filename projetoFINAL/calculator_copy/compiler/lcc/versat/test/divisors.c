#include "printi.h"
int divisors(int n) {
	int i = 2, divs = 0;
	while (i <= n/2) {
		if (n % i == 0) {
			printi(i, 10);
			putchar('\n');
			divs++;
		}
		i++;
	}
	return divs;
}

int main() { return divisors(144); }
