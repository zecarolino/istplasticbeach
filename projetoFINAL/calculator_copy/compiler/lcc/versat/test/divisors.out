VCD info: dumpfile xtop.vcd opened for output.
2
3
4
6
8
9
12
16
18
24
36
48
72
=
Catched Trap at time  137036 clock cycles:
